src/Visualisation/ConfigurationMenuController.java

```diff
@@ -0,0 +1,205 @@
+package Visualisation;
+
+import Island.IslandParams;
+import javafx.fxml.FXML;
+import javafx.scene.control.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConfigurationMenuController {
+    int[][][] TOPOLOGY = new int[][][]{
+            {
+                    {0, 1, 1, 1},
+                    {1, 0, 1, 1},
+                    {1, 1, 0, 1},
+                    {1, 1, 1, 0}
+            },
+            {
+                    {0, 1, 0, 0},
+                    {0, 0, 1, 0},
+                    {0, 0, 0, 1},
+                    {1, 0, 0, 0}
+            },
+            {
+                    {0, 1, 0, 0},
+                    {1, 0, 0, 0},
+                    {1, 0, 0, 1},
+                    {0, 1, 1, 0}
+            },
+            {
+                    {0, 1, 1, 0},
+                    {1, 0, 0, 1},
+                    {1, 0, 0, 1},
+                    {0, 1, 1, 0}
+            }
+    };
+    private MainApp mainApp;
+    private TopologyParams topology;
+    private List<IslandParams> islands;
+
+    ToggleGroup group = new ToggleGroup();
+
+    @FXML
+    private RadioButton unidirectionalCircleRadioButton, eachOfEachRadioButton, ladderRadioButton, bidirectionalCircleRadioButton;
+
+    @FXML
+    private TextField generationsTextField;
+
+    @FXML
+    private SplitMenuButton functionsSplitMenuButton;
+
+    @FXML
+    private TextField minimumDomainTextField, maximumDomainTextField;
+
+    @FXML
+    private TextField startingPopulation1TextField, startingPopulation2TextField,
+            startingPopulation3TextField, startingPopulation4TextField;
+
+    @FXML
+    private Slider basicCrossover1Slider, basicCrossover2Slider,
+            basicCrossover3Slider, basicCrossover4Slider;
+
+    @FXML
+    private Slider basicMutation1Slider, basicMutation2Slider,
+            basicMutation3Slider, basicMutation4Slider;
+
+    @FXML
+    private Slider singlePointCrossover1Slider, singlePointCrossover2Slider,
+            singlePointCrossover3Slider, singlePointCrossover4Slider;
+
+    @FXML
+    private Slider singlePointMutation1Slider, singlePointMutation2Slider,
+            singlePointMutation3Slider,singlePointMutation4Slider;
+
+    @FXML
+    private Slider migrationRate1Slider, migrationRate2Slider,
+            migrationRate3Slider, migrationRate4Slider;
+
+    @FXML
+    private TextField tournamentSize1TextField, tournamentSize2TextField,
+            tournamentSize3TextField, tournamentSize4TextField;
+
+    @FXML
+    private CheckBox elitism1CheckBox, elitism2CheckBox,
+            elitism3CheckBox, elitism4CheckBox;
+
+    @FXML
+    private ProgressBar progressBar;
+
+    @FXML
+    private Button startButton, cancelButton;
+
+    @FXML
+    private void handleStartButtonAction() {
+        if(validate_configuration()) {
+            setParams();
+        }
+
```

Comment:
Chodzi o to że ta funkcja zostaje wywoływana gdy naciśniesz przycisk i do takiej akcji jest przypisana. Jak wrzucisz ją do initialize() to podczas odpalania wywołujesz tą funkcje, więc od razu się cały form chce podsumować(zachowanie niepożądane przecież nie wprowadziliśmy żadnych parametrów), nie masz pewności że wszystkie elementy zostały zainicjalizowane, to sypie błędami :) nie wiem czy to wystarczająco wyjaśnia...